### PR TITLE
[Relay/TOPI][Op] Add batch_matmul in relay and TOPI

### PR DESCRIPTION
**Pull request by icemelon9**
This commit aims to add batch_dot op in both TOPI and Relay.
- [x] Add batch_dot op in Relay
- [x] Add x86 schedule in TOPI
- [x] Add GPU schedule in TOPI

### CODE REVIEW

File: topi/python/topi/x86/nn.py [hunk 1]
```diff
@@ -239,3 +236,50 @@ def _default_dense_nopack_config(cfg, M, N, K):
     cfg["tile_k"] = SplitEntity([K // tilek_bn, tilek_bn])
     cfg["tile_x"] = SplitEntity([N, 1])
     cfg["tile_y"] = SplitEntity([1, M])
+
+
+@generic.schedule_batch_dot.register(["cpu"])
+def schedule_batch_dot(outs):
+    """Schedule for softmax
```
Comment (yinghai) [inline topi/python/topi/x86/nn.py]: typo `softmax`

**Comment by yinghai** [comment]
Given the description of the computation, batch_matmul might be a more general name?

**Comment by icemelon9** [comment]
@yinghai Yes, I agree. I'll update it in the new commit.

**Comment by icemelon9** [comment]
@yinghai @Laurawly @Huyuwei Could you help review this PR? Thanks.

File: topi/tests/python/test_topi_batch_matmul.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        f(a, b, c)
+        tvm.testing.assert_allclose(c.asnumpy(), c_np, rtol=1e-5)
+
+    for device in get_all_backend():
+        check_device(device)
+
+def test_batch_matmul():
+    verify_batch_matmul(1, 16, 16, 32)
+    verify_batch_matmul(5, 16, 16, 32)
+    verify_batch_matmul(5, 16, 20, 32)
```
Comment (Laurawly) [inline topi/tests/python/test_topi_batch_matmul.py]: Maybe more test cases with batch size > 16?

Comment (icemelon9) [reply topi/tests/python/test_topi_batch_matmul.py]: Added test case for larger batch size.

**Comment by Laurawly** [comment]
@yzhliu @Huyuwei Could you add your reviews or approve this PR?

**Comment by icemelon9** [comment]
@ZihengJiang test_quantize_pass failed. Could you set rtol to a higher threshold?

**Comment by tqchen** [comment]
@icemelon9 can you try rebase and try to get it in?

**Comment by icemelon9** [comment]
Sure. But I'm still waiting on @ZihengJiang to fix the quantize pass test case.

**Comment by Laurawly** [comment]
Thanks @icemelon9 @yinghai @ZihengJiang , this is now merged.

**Comment by apivovarov** [comment]
I some cases relay.build hangs. The issue caused by `topi/python/topi/x86/dense.py` Line 194`.

More info https://discuss.tvm.ai/t/relay-build-hangs-for-5d-max-pool3d-reshape-matmul/5398

**Pull request merged**